### PR TITLE
Add new Dockerfile.proxy 

### PR DESCRIPTION
**Pull request by spagochitarra**
Added a new Dockerfile.proxy to create the container when building it happens behind a proxy . 
Modify the src/README.md file to describe it.

### CODE REVIEW

File: build/Dockerfile.proxy [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    lcov
+RUN pip3 install httpserver netaddr
+
+RUN echo "--- installing grpc dependencies ---" && \
+    apt-get install -y \
+    cmake libssl-dev \
+    autoconf git pkg-config \
+    automake libtool make g++ unzip 
+
+RUN echo "--- installing git dependencies ---" && \
```
Comment (er1cthe0ne) [inline build/Dockerfile.proxy]: Thanks for finding this solution for environment with proxy enabled.

File: src/README.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ Source code folder:
 - net_config: Library for configurating host networking components
 - proto3: Auto generated source codes and library for proto3 scheme for communication with Alcor Controllers
 - transit_rpc: Library for RPC interface with transit daemon
-- test: Unit and intergration test code
+- test: Unit and integration test code
```
Comment (er1cthe0ne) [inline src/README.md]: :) thanks for catching this.

File: src/README.md [hunk 3]
```diff
@@ -39,3 +39,62 @@ You should be ready to run the executable:
 ```Shell
 root@ca62b6feec63:/mnt/host/code/AlcorControlAgent# ./build/bin/AlcorControlAgent
 ```
+
+# Build the container while behind a proxy
+
+If the docker installation environment is behind proxy the Docker.proxy file needs
```
Comment (er1cthe0ne) [inline src/README.md]: Dockerfile.proxy

Comment (spagochitarra) [reply src/README.md]: Fixed. Thanks for catching this.

Comment (er1cthe0ne) [reply src/README.md]: You are welcome.

**Pull request merged**